src/main/java/org/jenkinsci/plugins/junitrealtimetestreporter/RealtimeJUnitStep.java

```diff
@@ -168,96 +174,99 @@ public boolean start() throws Exception {
             );
             AbstractRealtimeTestResultAction.saveBuild(r);
             context.newBodyInvoker().withCallback(new Callback(id, archiver)).start();
-            return false;
         }
 
-        private static final long serialVersionUID = 1;
-
-    }
+        @SuppressFBWarnings(value = "SE_BAD_FIELD", justification = "Handled by 'Pickler' below")
+        private final class Callback extends BodyExecutionCallback {
 
-    @SuppressFBWarnings("SE_BAD_FIELD") // FIXME
-    static class Callback extends BodyExecutionCallback {
```

Comment:
`Callback` could be serialized to `program.dat` so this will break any build running across the upgrade. I think you need to deprecate the original but keep it.